src/components/Contacts/ContactsHeader/ContactsHeader.test.tsx

```diff
@@ -200,8 +200,9 @@ describe('ContactsHeader', () => {
     userEvent.type(textbox, searchText);
 
     expect(toggleFilterPanel).not.toHaveBeenCalled();
-    setTimeout(() => {
-      expect(onSearchTermChanged).toHaveBeenCalledWith(searchText);
-    }, 500);
+
+    await new Promise((resolve) => setTimeout(resolve, 500));
+
+    expect(onSearchTermChanged).toHaveBeenCalledWith(searchText);
```

Comment:
This is better and actually runs the expect but I think using Jest's fake timers would be better so the test doesn't have to wait half a second doing nothing.
https://jestjs.io/docs/timer-mocks#advance-timers-by-time

Reply:
Hmm from what I can tell, Jest fake timers don't work with lodash/debounce.  I was looking at this comment thread (https://github.com/facebook/jest/issues/3465), it looks like the options are to either use real timers along with `setTimeout` https://github.com/facebook/jest/issues/3465#issuecomment-335160011 or mocking lodash https://github.com/facebook/jest/issues/3465#issuecomment-351186130.

Reply:
Ah good to know. This is fine.